### PR TITLE
feature: added DTCWT operator

### PR DESCRIPTION
**Pull request by mrava87**
This PR introduced the `DTCWT` operator in the signal processing submodule that implements the Dual-Tree Complex Wavelet Transform

### CODE REVIEW

File: pylops/utils/deps.py [hunk 1]
```diff
@@ -49,6 +51,23 @@ def devito_import(message):
...
+            import dtcwt  # noqa: F401
+
+            dtcwt_message = None
+        except Exception as e:
+            dtcwt_message = f"Failed to import dtcwt (error:{e})."
+    else:
+        dtcwt_message = (
+            f"Dtcwt not available. "
+            f"In order to be able to use "
+            f'{message} run "pip install devito".'
```
Comment (marcusvaltonen) [inline pylops/utils/deps.py:66]: pip install dcwt

File: pylops/signalprocessing/__init__.py [hunk 2]
```diff
@@ -24,7 +24,7 @@
     DWT                             One dimensional Wavelet operator.
     DWT2D                           Two dimensional Wavelet operator.
     DCT                             Discrete Cosine Transform.
-    Seislet                         Two dimensional Seislet operator.
```
Comment (marcusvaltonen) [inline pylops/signalprocessing/__init__.py:27]: Is Seislet removed?

File: pylops/utils/deps.py (hunk 1 above)
Comment (mrava87) [reply pylops/utils/deps.py:66]: Ops good point

**Comment by mrava87** [comment]
@marcusvaltonen unfortunately this may never make into pylops… dcwt has some issues with newer versions of numpy, there is even a PR open but the project seems stalled :( https://github.com/rjw57/dtcwt/pull/144

**Comment by mrava87** [comment]
@marcusvaltonen seems like there has been some work on the dtcwt library and it is now compatible with the latest version of numpy... I am about to close this and make a new clean PR to include the DTCWT operator in pylops

**Pull request closed without merging**